### DISCUSSION TITLE
How to access OnDrag events inside of draggable

### DISCUSSION THREAD

**Discussion by Logaqm** [clauderic/dnd-kit#268] [Q&A] [answered]
I have a variable list of draggables im rendering, how can i access the onDrag events in the draggables component, rather than on the dndcontext? It would be much easier if they were able to listen to this event and update their own transform.

**Comment by clauderic** [comment, marked as answer]
The `useDndMonitor` hook can be used within components wrapped in a DndContext provider to monitor the different drag and drop events that happen for that DndContext.

```ts
import {DndContext, useDndMonitor} from '@dnd-kit/core';

function App() {
  return (
    <DndContext>
      <Component />
    </DndContext>
  );
}

function Component() {
  // Monitor drag and drop events that happen on the parent `DndContext` provider
  useDndMonitor({
    onDragStart(event) {},
    onDragMove(event) {},
    onDragOver(event) {},
    onDragEnd(event) {},
    onDragCancel(event) {},
  });
}
```

https://docs.dndkit.com/api-documentation/context-provider/use-dnd-monitor

**Reply by Logaqm** [reply]
Wow, I completely missed that in the documentation.
